### PR TITLE
clean up proficiency variable names

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Clean up names across the codebase that refer to proficiency rather than frequency of skill demonstration.

NOTE: we still use the language of proficiency in our diagnostic reports and our standards reports (both for teachers and for admins). Variables used for these reports were therefore left unchanged.

## WHY
We want our variable naming to reflect what's displayed to users in order to avoid confusion.

## HOW
Lots of global searches, updates, and manual testing.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Clean-up-Task-Update-codebase-to-move-away-from-variable-names-around-proficiency-vs-demonstrated--f53d3189c27549f99febaf95b7df288a?pvs=4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

File: services/QuillLMS/app/assets/stylesheets/pages/student_dashboard/student_profile_unit.scss [hunk 1]
```diff
@@ -86,13 +86,13 @@
         &.completed {
           background-color: #358fdf;
         }
-        &.proficient {
+        &.freqnetly-demonstrated-skill {
```
Comment (eadams17) [inline services/QuillLMS/app/assets/stylesheets/pages/student_dashboard/student_profile_unit.scss]: typo here

File: services/QuillLMS/app/assets/stylesheets/pages/student_results.scss [hunk 2]
```diff
@@ -69,13 +69,13 @@
...
               background-color: #4ea500;
             }
-            &.nearly-proficient {
+            &.sometimes-demonstrated-skill {
               background-color: #eb9911;
             }
-            &.not-yet-proficient {
+            &.rarely-demonstrated-skill {
               background-color: #e73030;
             }
```
Comment (eadams17) [inline services/QuillLMS/app/assets/stylesheets/pages/student_results.scss:80]: we should probably update all of these hex codes to be variables?

**Review by eadams17 — APPROVED** [review]
Thanks for cleaning this up!

File: services/QuillLMS/app/assets/stylesheets/pages/student_results.scss (hunk 2 above)
Comment (emilia-friedberg) [reply services/QuillLMS/app/assets/stylesheets/pages/student_results.scss:80]: These hex codes don't currently correspond to variables, and there isn't yet a correspondence for them in the mapping.

**Review by anathomical — APPROVED** [review]

**Pull request merged**